File: 2114/typescript/2114.ts
// tsc 2114.ts
// node 2114.js

function mostWordsFound(sentences: string[]): number {
	return sentences.reduce(
		(maxWords: number, sentence: string) =>
			Math.max(maxWords, sentence.split(" ").length),
		0
	);
}

console.log(
	mostWordsFound([
		"alice and bob love leetcode",
		"i think so too",
		"this is great thanks very much",
	])
);

console.log(
	mostWordsFound(["please wait", "continue to fight", "continue to win"])
);

File: 2114/python/2114.py
# python 2114.py

class Solution(object):
    def mostWordsFound(self, sentences):
        return max(len(sentence.split()) for sentence in sentences)

solution = Solution()

print(
    solution.mostWordsFound(
        [
            "alice and bob love leetcode",
            "i think so too",
            "this is great thanks very much",
        ]
    )
)

print(
    solution.mostWordsFound(
        [
            "please wait",
            "continue to fight",
            "continue to win",
        ]
    )
)
